imsim/skycat.py

```diff
@@ -0,0 +1,355 @@
+            LSST band, e.g., 'u', 'g', 'r', 'i', 'z', or 'y'
+        obj_types : list-like [None]
+            List or tuple of object types to render, e.g., ('star', 'galaxy').
+            If None, then consider all object types.
+        edge_pix : float [100]
+            Size in pixels of the buffer region around nominal image
+            to consider objects.
+        logger : logging.Logger
+            Logger object.
+        """
+        logger = galsim.config.LoggerWrapper(logger)
+        if obj_types is not None:
+            logger.warning(f'Object types restricted to {obj_types}')
+        self.file_name = file_name
+        self.wcs = wcs
+        self.band = band
+        sky_cat = skyCatalogs.open_catalog(file_name)
+        region = skyCatalogs.Box(*get_radec_limits(wcs, logger, edge_pix)[:4])
+        self.objects = sky_cat.get_objects_by_region(region,
+                                                     obj_type_set=obj_types)
+
+    def getNObjects(self):
+        """
+        Return the number of GSObjects to render, where each subcomponent
+        (e.g., bulge, disk, etc.) of each skyCatalog object is a distinct
+        GSObject.
+        """
+        return len(self.objects)
+
+    @property
+    def nobjects(self):
+        return self.getNObjects()
+
+    def getSED_info(self, skycat_obj, component):
+        """
+        Return the SED and magnorm value of the skyCatalog object
+        corresponding to the specified index.
+
+        Parameters
+        ----------
+        skycat_obj : skycatalogs.BaseObject
+            The skycatalogs object, e.g., star or galaxy, which can
+            have several subcomponents, e.g., disk, bulge, knots.
+        component : str
+            Name of the subcomponent.
+
+        Returns
+        -------
+        (galsim.SED, float)
+        """
+
+        wl, flambda, magnorm \
+            = skycat_obj.get_sed(component=component)
+        if np.isinf(magnorm):
+            # Galaxy subcomponents, e.g., bulge components, can
+            # have zero-valued SEDs.  The skyCatalogs code returns
+            # magnorm=inf for these components.
+            return None, magnorm
+        sed_lut = galsim.LookupTable(wl, flambda)
+        sed = galsim.SED(sed_lut, wave_type='nm', flux_type='flambda')
+        sed = sed.withMagnitude(0, self._bp500)
+        iAv, iRv, mwAv, mwRv = self.getDust(skycat_obj)
+
+        # TODO: apply internal extinction here
+
+        # Apply redshift.
+        if 'redshift' in skycat_obj.native_columns:
+            redshift = skycat_obj.get_native_attribute('redshift')
+            sed = sed.atRedshift(redshift)
+
+        # Apply Milky Way extinction
+        extinction = F19(Rv=mwRv)
+        wl = np.linspace(300, 1200, 901)
+        ext = extinction.extinguish(wl*u.nm, Av=mwAv)
+        spec = galsim.LookupTable(wl, ext)
+        mw_ext = galsim.SED(spec, wave_type='nm', flux_type='1')
+        sed = sed*mw_ext
+
+        sed = sed.thin()
```

Comment:
I had a comment before about removing this line, which seems to be gone now.  This is a very non-trivial calculation.  If we do it on every object, it will be faster to not do it. It's only useful if you do it once on and SED and then reuse that SED repeatedly.

I think what you want here is to use sed.wave_list rather than a new `np.linspace`, which you then have to thin.